### DISCUSSION TITLE
How to save BelongsToMany records from the handleRecordCreation function?

### DISCUSSION THREAD

**Discussion by rmitesh** [filamentphp/filament#6293] [Help] [answered]
I want to customize `handleRecordCreation` function, and attach user record manually?

Here is my function for the same...

```php

// ProjectScopes.php => it is a trait class

protected static function getSuperAdminUser( array $select = ['id'] ): User
{
    return User::role(User::ROLE_SUPER_ADMIN)->select($select)->first();
}
```

```php

// CreateProject.php

protected function handleRecordCreation(array $data): Model
{
    $data['created_by'] = auth()->id();
    $record = static::getModel()::create($data);

    $users = collect([static::getSuperAdminUser()]);
    
    $record->defaultAssignToUsers()->sync($users->pluck('id')->toArray());
    return $record;
}
```
Note:
- If I add `dd()` before the `return $record;` then data is saved in the table, otherwise is not working.

**Comment by rmitesh** [comment]
@danharrin I know the question is tricky, but have you get an chance to review this?

**Reply by danharrin** [reply]
I don't face enough information to be able to suggest anything, and I don't think it's necessary Filament-related

**Reply by rmitesh** [reply, marked as answer]
Hi @danharrin , you were right. after all pushing my head, finally I found it!... 😂

I just realized that I missing something here. I forgot to add 2nd parameters in the `sync()`. it's for prevent to detaching existing one.
`->sync($users->pluck('id')->toArray(), true)` 

And it's worked!...
